### ISSUE TITLE
How to a Autotrain Seq2Seq ? 

### ISSUE THREAD

**Issue by Hopeforzbest** [huggingface/autotrain-advanced#376]
Hi everyone , I'm trying to finetune a Helsinki-NLP/opus-mt-tc-big-ar-en on local arabic of morocco which is called Daraija Arabic , the problem is that I'm unable to use Autotrain I keep getting 500 error code 
![Screenshot 2023-12-07 011848](https://github.com/huggingface/autotrain-advanced/assets/112639221/ece3ee15-9f89-44ff-bf51-c5231f1858e7)
![Screenshot 2023-12-07 011912](https://github.com/huggingface/autotrain-advanced/assets/112639221/2dea03ae-afcd-4e86-a7b3-d175ff6bc555)
[output.csv](https://github.com/huggingface/autotrain-advanced/files/13593069/output.csv)
FYI : I didnt modify Training Parameters (find params to copy-paste [here] area so I dont know if its necessary

**Comment by Hopeforzbest** [comment]
===== Application Startup at 2023-12-06 22:31:45 =====


==========
== CUDA ==
==========

CUDA Version 12.1.1

Container image Copyright (c) 2016-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.

This container image and its contents are governed by the NVIDIA Deep Learning Container License.
By pulling and using the container, you accept the terms and conditions of this license:
https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license

A copy of this license is made available in this container at /NGC-DL-CONTAINER-LICENSE for your convenience.

WARNING: The NVIDIA Driver was not detected.  GPU functionality will not be available.
   Use the NVIDIA Container Toolkit to start this container with GPU support; see
   https://docs.nvidia.com/datacenter/cloud-native/ .

INFO:     Will watch for changes in these directories: ['/app']
WARNING:  "workers" flag is ignored when reloading is enabled.
INFO:     Uvicorn running on http://0.0.0.0:7860 (Press CTRL+C to quit)
INFO:     Started reloader process [35] using StatReload
> INFO    Authenticating user...
> WARNING Parameters not supplied by user and set to default: merge_adapter, token, repo_id, lr, evaluation_strategy, logging_steps, save_total_limit, weight_decay, model_max_length, model_ref, batch_size, trainer, max_grad_norm, project_name, username, lora_alpha, lora_r, valid_split, warmup_ratio, train_split, lora_dropout, data_path, optimizer, dpo_beta, save_strategy, model, disable_gradient_checkpointing, seed, auto_find_batch_size, text_column, gradient_accumulation, use_flash_attention_2, prompt_text_column, push_to_hub, add_eos_token, rejected_text_column, scheduler
> WARNING Parameters not supplied by user and set to default: scheduler, train_split, token, target_column, data_path, optimizer, repo_id, lr, max_seq_length, fp16, save_total_limit, logging_steps, evaluation_strategy, epochs, weight_decay, save_strategy, log, batch_size, model, seed, auto_find_batch_size, max_grad_norm, text_column, gradient_accumulation, username, project_name, push_to_hub, valid_split, warmup_ratio
> WARNING Parameters not supplied by user and set to default: scheduler, train_split, token, target_column, data_path, optimizer, repo_id, lr, fp16, save_total_limit, logging_steps, evaluation_strategy, epochs, weight_decay, save_strategy, log, batch_size, model, seed, auto_find_batch_size, max_grad_norm, gradient_accumulation, project_name, image_column, push_to_hub, valid_split, warmup_ratio
> WARNING Parameters not supplied by user and set to default: token, repo_id, validation_prompt, xformers, epochs, rank, resume_from_checkpoint, train_text_encoder, class_image_path, lr_power, adam_weight_decay, max_grad_norm, num_validation_images, tokenizer, adam_beta1, resolution, allow_tf32, local_rank, username, project_name, tokenizer_max_length, prior_generation_precision, pre_compute_text_embeddings, num_cycles, num_class_images, dataloader_num_workers, text_encoder_use_attention_mask, center_crop, adam_beta2, revision, warmup_steps, checkpoints_total_limit, validation_images, bf16, prior_loss_weight, image_path, model, sample_batch_size, seed, class_labels_conditioning, xl, checkpointing_steps, validation_epochs, logging, class_prompt, prior_preservation, push_to_hub, scale_lr, adam_epsilon, scheduler, use_8bit_adam
> WARNING Parameters not supplied by user and set to default: token, repo_id, lr, max_seq_length, fp16, evaluation_strategy, logging_steps, save_total_limit, epochs, weight_decay, batch_size, max_grad_norm, username, project_name, lora_alpha, lora_r, valid_split, warmup_ratio, train_split, target_column, max_target_length, use_int8, data_path, optimizer, lora_dropout, save_strategy, use_peft, target_modules, model, seed, auto_find_batch_size, text_column, gradient_accumulation, push_to_hub, scheduler
> WARNING Parameters not supplied by user and set to default: train_split, token, id_column, categorical_imputer, data_path, repo_id, time_limit, numerical_columns, model, seed, task, categorical_columns, numerical_imputer, numeric_scaler, target_columns, username, project_name, num_trials, push_to_hub, valid_split
INFO:     Started server process [37]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     10.16.20.172:53634 - "GET / HTTP/1.1" 200 OK
INFO:     10.16.20.172:53634 - "GET /logo.png HTTP/1.1" 200 OK
> INFO    Task: llm:sft
INFO:     10.16.18.44:39542 - "GET /params/llm%3Asft HTTP/1.1" 200 OK
> INFO    hardware: A10G Large
INFO:     10.16.18.44:25742 - "POST /create_project HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/app/env/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/app/env/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/fastapi/applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/app/env/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/app/env/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/app/env/lib/python3.10/site-packages/fastapi/routing.py", line 274, in app
    raw_response = await run_endpoint_function(
  File "/app/env/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/app/src/autotrain/app.py", line 248, in handle_form
    dset = AutoTrainDataset(
  File "<string>", line 12, in __init__
  File "/app/src/autotrain/dataset.py", line 198, in __post_init__
    self.train_df, self.valid_df = self._preprocess_data()
  File "/app/src/autotrain/dataset.py", line 207, in _preprocess_data
    train_df.append(pd.read_csv(file))
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 948, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 617, in _read
    return parser.read(nrows)
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 1748, in read
    ) = self._engine.read(  # type: ignore[attr-defined]
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/c_parser_wrapper.py", line 234, in read
    chunks = self._reader.read_low_memory(nrows)
  File "parsers.pyx", line 843, in pandas._libs.parsers.TextReader.read_low_memory
  File "parsers.pyx", line 904, in pandas._libs.parsers.TextReader._read_rows
  File "parsers.pyx", line 879, in pandas._libs.parsers.TextReader._tokenize_rows
  File "parsers.pyx", line 890, in pandas._libs.parsers.TextReader._check_tokenize_status
  File "parsers.pyx", line 2058, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: Expected 2 fields in line 21, saw 3

> INFO    Task: seq2seq
INFO:     10.16.18.44:16455 - "GET /params/seq2seq HTTP/1.1" 200 OK
> INFO    hardware: A10G Large
INFO:     10.16.20.172:47809 - "POST /create_project HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/app/env/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/app/env/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/fastapi/applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/app/env/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/app/env/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/app/env/lib/python3.10/site-packages/fastapi/routing.py", line 274, in app
    raw_response = await run_endpoint_function(
  File "/app/env/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/app/src/autotrain/app.py", line 273, in handle_form
    dset = AutoTrainDataset(
  File "<string>", line 12, in __init__
  File "/app/src/autotrain/dataset.py", line 198, in __post_init__
    self.train_df, self.valid_df = self._preprocess_data()
  File "/app/src/autotrain/dataset.py", line 207, in _preprocess_data
    train_df.append(pd.read_csv(file))
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 948, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 617, in _read
    return parser.read(nrows)
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 1748, in read
    ) = self._engine.read(  # type: ignore[attr-defined]
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/c_parser_wrapper.py", line 234, in read
    chunks = self._reader.read_low_memory(nrows)
  File "parsers.pyx", line 843, in pandas._libs.parsers.TextReader.read_low_memory
  File "parsers.pyx", line 904, in pandas._libs.parsers.TextReader._read_rows
  File "parsers.pyx", line 879, in pandas._libs.parsers.TextReader._tokenize_rows
  File "parsers.pyx", line 890, in pandas._libs.parsers.TextReader._check_tokenize_status
  File "parsers.pyx", line 2058, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: Expected 2 fields in line 21, saw 3

INFO:     10.16.20.172:17946 - "GET / HTTP/1.1" 200 OK
INFO:     10.16.20.172:17946 - "GET /logo.png HTTP/1.1" 200 OK
> INFO    Task: llm:sft
INFO:     10.16.20.172:17946 - "GET /params/llm%3Asft HTTP/1.1" 200 OK
> INFO    hardware: A10G Large
INFO:     10.16.20.172:46007 - "POST /create_project HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/app/env/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/app/env/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/fastapi/applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/app/env/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/app/env/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/app/env/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/app/env/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/app/env/lib/python3.10/site-packages/fastapi/routing.py", line 274, in app
    raw_response = await run_endpoint_function(
  File "/app/env/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/app/src/autotrain/app.py", line 273, in handle_form
    dset = AutoTrainDataset(
  File "<string>", line 12, in __init__
  File "/app/src/autotrain/dataset.py", line 198, in __post_init__
    self.train_df, self.valid_df = self._preprocess_data()
  File "/app/src/autotrain/dataset.py", line 207, in _preprocess_data
    train_df.append(pd.read_csv(file))
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 948, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 617, in _read
    return parser.read(nrows)
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/readers.py", line 1748, in read
    ) = self._engine.read(  # type: ignore[attr-defined]
  File "/app/env/lib/python3.10/site-packages/pandas/io/parsers/c_parser_wrapper.py", line 234, in read
    chunks = self._reader.read_low_memory(nrows)
  File "parsers.pyx", line 843, in pandas._libs.parsers.TextReader.read_low_memory
  File "parsers.pyx", line 904, in pandas._libs.parsers.TextReader._read_rows
  File "parsers.pyx", line 879, in pandas._libs.parsers.TextReader._tokenize_rows
  File "parsers.pyx", line 890, in pandas._libs.parsers.TextReader._check_tokenize_status
  File "parsers.pyx", line 2058, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: Expected 2 fields in line 21, saw 3

INFO:     10.16.20.172:57414 - "GET /robots.txt HTTP/1.1" 404 Not Found
INFO:     10.16.18.44:7777 - "GET / HTTP/1.1" 200 OK
INFO:     10.16.20.172:1615 - "GET /logo.png HTTP/1.1" 200 OK
INFO:     10.16.20.172:49738 - "GET / HTTP/1.1" 200 OK
INFO:     10.16.34.18:1883 - "GET /logo.png HTTP/1.1" 200 OK
> INFO    Task: llm:sft
INFO:     10.16.34.18:1883 - "GET /params/llm%3Asft HTTP/1.1" 200 OK

**Comment by Hopeforzbest** [comment]
===== Build Queued at 2023-12-06 22:31:36 / Commit SHA: 71571d6 =====

--> FROM docker.io/huggingface/autotrain-advanced:latest@sha256:a160d4e1549a5ab6b98ecac68a9f67cd5d6108c70f121ff13022f92191a5f68b
DONE 0.0s

--> Pushing image
DONE 2.3s

--> Exporting cache
DONE 0.1s

**Comment by abhishekkrthakur** [comment]
Let's keep question limited to one single channel to avoid confusion :)
your file doesnt look like a proper CSV. some rows have commas in the sentences that's causing it to believe it has more than two columns. you can fix it by using proper quoting.

**Comment by abhishekkrthakur** [comment]
its just a guess from: pandas.errors.ParserError: Error tokenizing data. C error: Expected 2 fields in line 21, saw 3 :)

**Comment by abhishekkrthakur** [comment]
please make sure your project name is unique and there are no other repositories in your huggingface account with the same name: no datasets, models or spaces. i generally use the randomly generated name and then rename the model later if needed.

**Comment by Hopeforzbest** [comment]
I used a unique name : lm_logits = self.lm_head(outputs[0]) + self.final_logits_bias
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 6.48 GiB. GPU 0 has a total capacty of 21.99 GiB of which 6.47 GiB is free. Process 43024 has 15.51 GiB memory in use. Of the allocated memory 13.89 GiB is allocated by PyTorch, and 1.31 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF , is it because I use a free GPU ?
